promise_button.js

```diff
@@ -75,17 +98,62 @@ angular.module('ngPromiseButton', [])
           }
 
           var promise = scope.promiseButton();
-          scope.state = 'progress';
+    		  if(promise){
+    			
+    			promise = promise.$promise ? promise.$promise : promise; // handle ngResource
+                
+    			if(scope.promiseSubmit){
```

Comment:
Not getting purpose of this. What is the difference from just setting button as `submit` type in the parent scope?

```
<form>
...
<button type='submit' promise-button='submit()'>Save</button>
</form>
```